tests/test_cli.py

```diff
@@ -138,6 +141,15 @@ def test_add_remove_profile():
     assert result.exit_code == 1
     assert "Profile with name 'new-profile' does not exist." in result.output
 
+    # Remove second-profile with reset (`--purge` option)
+    result: Result = runner.invoke(
+        cli.cli,
+        ["profile", "remove", "--purge", "second-profile"],
+        input="y\nsecond-profile\n",
+    )
+    assert result.exit_code == 0
+    assert "Please enter the name of the profile to continue" in result.output
```

Comment:
I tried to move `get_volume` out and reuse it for this purge option, but give up since I realize it break the good design of using `instance` fixture as the class scope and cleaned up it after the test was finalized.

Reply:
Thanks for taking a look. I think my idea was more simple. Instead of launching the whole lifecycle, you would  _create_ the docker volumes manually in the fixture.

Reply:
I have a second try and found the problem is the profile and container serve two different cases. When user run `aiidalab-launch profile` it only create the profile but did nothing with container and the volume. It is when the profile first time used the container and its volumes are created. 

If you try with `-vvv` with this pr branch, you'll find that before the newly creating profile first time running, running with `--purge` will go through `reset` which manipulate docker container directly and hit the code path https://github.com/aiidalab/aiidalab-launch/blob/33c7a26fe74ffcbea7676fa98f02769b07bc631a/aiidalab_launch/instance.py#L239-L242

It won't cause any issues because it logs to the debug level. I am now not very confident in myself it is a good way to implement the purge here which mixes the profile and container controlling part of the code.
From what you are asking here, it is useless to create docker volume fixture because in this test, only the profile is created but not the container. So it has to go to the real life cycle which has the container running with volumes created.

Reply:
In fact, I think the profile concept is not very useful to end user. If we regard `aiidalab-launch` as a wrapper of a combination of multiple docker commands, it maybe makes sense to have the CLI command match with the docker command which are: 
- `docker run` -> `aiidalab-launch run`: to start from an image. (this will be the combination of current `aiidalab-launch profile add` + `aiidalab-launch profile start`)
- `docker stop` -> `aiidalab-launch stop`: stop the container, it is the same from what it is now.
- `docker rm` -> `aiidalab-launch rm`: remove the container (for aiidalab-launch case also remove the volume), this is then what I was going to achieve in this PR. It is the combination of the current `aiidalab-launch reset` and `aiidalab-launch profile remove`.

Reply:
Hi @danielhollas, can you give this a look? We can then decide we halt this PR or continue it with brining some mixture of profile and container commands.